src/core/net/srp_client.cpp

```diff
@@ -901,6 +901,12 @@ Error Client::ReadOrGenerateKey(Crypto::Ecdsa::P256::KeyPair &aKeyPair)
         }
     }
 
+#if OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE
+    aKeyPair.Init(kSrpEcdsaKeyRef);
+#else
+    aKeyPair.Init();
+#endif
+
     SuccessOrExit(error = aKeyPair.Generate());
     IgnoreError(Get<Settings>().Save<Settings::SrpEcdsaKey>(aKeyPair));
```

Comment:
Follow up from previous comment: How do we deal with this code here under `OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE`?

We save the newly generated key pair in non-volatile settings (in `Client::ReadOrGenerateKey()` method).
Under `PLATFORM_KEY_REFERENCES_ENABLE` the `Generate()` will not give us the literal key anymore and save it in PSA:
```
* If OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE config is enabled, the
 * generated keypair is stored inside PSA ITS and mDerBytes in @p aKeyPair is not updated
 * with the literal key.
 * 
 ```
I think we need to how we save/read the "SRP ECDSA key" in/from `Settings` under `PLATFORM_KEY_REFERENCES_ENABLE`?